src/VisualStudio/Core/Def/LanguageService/AbstractPackage.cs

```diff
@@ -27,16 +27,20 @@ internal IComponentModel ComponentModel
 
     protected override async Task InitializeAsync(CancellationToken cancellationToken, IProgress<ServiceProgressData> progress)
     {
-        await base.InitializeAsync(cancellationToken, progress).ConfigureAwait(true);
+        // Should only be called from a threadpool. Opinionated, as package load sequence thread switches are impactful.
+        Contract.ThrowIfTrue(JoinableTaskFactory.Context.IsOnMainThread);
```

Comment:
We inline you on UI thread if sync blocked so this will fire.

Reply:
This is a base class, the derived classes ensure we aren't on the main thread

Reply:
In general, we don't want packages to make that decision; we inline you on the UI thread because its not doing anything, if you queue to the thread pool you will now be affected by latency of the thread pool, instead of just immediately on the thread that is currently blocked on you,